eta/core/storage.py

```diff
@@ -58,6 +60,35 @@
 logging.getLogger("paramiko").setLevel(logging.ERROR)
 
 
+def google_cloud_api_retry(func):
```

Comment:
I bet this could also be applied to `GoogleDriveStorageClient`

Reply:
I just applied it to all functions in google drive, noting that it only catches specific errors and otherwise does nothing, so it's pretty safe to just apply it anywhere :)

Reply:
Did you actually test to see if the Google Drive API will raise errors from these namespaces?

Reply:
Also, importantly, some of these functions are not purely google API calls. They may perform multiple. For example, if you retry an entire function that tries to upload a list of files in series when the last one fails, then you'll get multiple copies of some files.

It's only safe to retry single API calls. Or actions that are otherwise idempotent.